### ISSUE TITLE
padded-blocks docs claim wrong default configuration

### ISSUE THREAD

**Issue by deniseli** [eslint/eslint#7126]
**Tell us about your environment**
Chrome!

* **ESLint Version:** 3.4.0
* **Node Version:** 6.5.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**What did you do? Please include the actual source code causing the issue.**
config:
`"padded-blocks": 2,`

js that was linted:
```
/* eslint no-unused-expressions: 0 */

import DomUtil from '../../js/visualizations/DomUtil';

describe('$SUITE$ DomUtil', function() {
  describe('strSize', function() {
    it('dimensions should increase with font size', function() {
      let width = 0;
      let height = 0;

      for (let i = 1; i < 20; i += 2) {
        const dims = DomUtil.strSize('test-string', i);
        expect(dims.width).to.be.greaterThan(width);
        expect(dims.height).to.be.greaterThan(height);
        width = dims.width;
        height = dims.height;
      }
    });
  });
});
```

**What did you expect to happen?**
Since http://eslint.org/docs/rules/padded-blocks says "always" is the default, I assumed padded-blocks would run with "always", but instead it ran with "never".

**Comment by platinumazure** [comment]
Can you try with the default parser?

Also, can you please edit the post to include the actual source code causing the issue?

A quick look at the source code shows that it does use "always" by default. There may be a bug here, but we don't have enough information to know for sure.

**Comment by deniseli** [comment]
Sure! Just updated the post. 

I got the same result with the default parser. Basically, when I explicitly set the config to `"padded-blocks": [2, "always"],` I get the following errors. When I don't, or when I set it to "never", I don't get errors.
```
   5:40  error  Block must be padded by blank lines  padded-blocks
   6:34  error  Block must be padded by blank lines  padded-blocks
   7:64  error  Block must be padded by blank lines  padded-blocks
  11:39  error  Block must be padded by blank lines  padded-blocks
  17:7   error  Block must be padded by blank lines  padded-blocks
  18:5   error  Block must be padded by blank lines  padded-blocks
  19:3   error  Block must be padded by blank lines  padded-blocks
  20:1   error  Block must be padded by blank lines  padded-blocks
```

**Comment by platinumazure** [comment]
Oh! Fascinating! I agree that looks like a bug. However, I can't reproduce in the online demo.

Here's what I tried (I took a subset of your code), maybe you can tell me if I've done something wrong?

/* eslint-env es6 */
/* eslint padded-blocks:2 */

```js
describe('$SUITE$ DomUtil', function() {
  describe('strSize', function() {
    it('dimensions should increase with font size', function() {
      let width = 0;
      let height = 0;

      for (let i = 1; i < 20; i += 2) {
        const dims = DomUtil.strSize('test-string', i);
        expect(dims.width).to.be.greaterThan(width);
        expect(dims.height).to.be.greaterThan(height);
        width = dims.width;
        height = dims.height;
      }
    });
  });
});
```

**Comment by deniseli** [comment]
Weird, that looks right. Here's the rest of my config, minus the rule configurations. Is anything non-standard?

```

{ 
  "extends": "airbnb-base",
  "parser": "babel-eslint",
  "plugins": [
    "react"
  ],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "env": {
    "es6": true,
    "browser": true,
    "node": true,
    "amd": false,
    "mocha": true,
    "jasmine": false
  },
  "ecmaFeatures": {
    "arrowFunctions": true,
    "blockBindings": true,
    "classes": true,
    "defaultParams": true,
    "destructuring": true,
    "forOf": true,
    "generators": false,
    "modules": true,
    "objectLiteralComputedProperties": true,
    "objectLiteralDuplicateProperties": false,
    "objectLiteralShorthandMethods": true,
    "objectLiteralShorthandProperties": true,
    "spread": true,
    "superInFunctions": true,
    "templateStrings": true,
    "jsx": true
  },
  "globals": {
    "sinon": true,
    "expect": true,
    "assert": true,
    "GridStackUI": true,
    "jQuery": true
  },
  "rules": {
```

**Comment by platinumazure** [comment]
Ah, I didn't realize you were extending from a config. This is why we like the whole config :smile:

Our config extension process has one interesting bit of behavior: If a user extends a config with *just* an error level, then everything else about the base config for that rule is intact. This is intentionally done in case a user wants to use a rule and only change its error level, but otherwise leave the config intact (if we did a full overwrite of the config, then the user would have to write the full config every time they make any change to a rule while extending a config).

In your case, eslint-config-airbnb has padded-blocks [set to never by default](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/style.js#L272).

One thing you can try is to specify the error level as an array: `"padded-blocks": [2]`, that might trigger a config overwrite. If that doesn't work, unfortunately you'll have to use the full config: `"padded-blocks": [2, "always"]`.

Hope that helps!

**Comment by deniseli** [comment]
Ah, interesting! That was very helpful to learn. Thanks!

**Comment by platinumazure** [comment]
@deniseli Please do let me know if the array (`[2]`) worked for you-- if not, I might open an issue to see if we can support both cases. Thanks!

**Comment by deniseli** [comment]
@platinumazure Whoops - using the array didn't work. I'll just use the full config in this case, but supporting the list case *would* be cool!

**Comment by platinumazure** [comment]
@deniseli Thank you for testing that for me! I'll see if I can get the team on board to support different behavior for `2` vs `[2]` overrides.
